src/Compilers/CSharp/Portable/Compilation/MemberSemanticModel.cs

```diff
@@ -959,30 +959,75 @@ public override SyntaxTree SyntaxTree
             }
         }
 
-        internal override IOperation GetOperationWorker(CSharpSyntaxNode node, GetOperationOptions options, CancellationToken cancellationToken)
+        internal override IOperation GetOperationWorker(CSharpSyntaxNode node, CancellationToken cancellationToken)
         {
-            CSharpSyntaxNode bindableNode;
+            CSharpSyntaxNode bindingRoot = GetBindingRootOrInitializer(node);
 
-            BoundNode lowestBoundNode;
-            BoundNode highestBoundNode;
-            BoundNode boundParent;
+            IOperation statementOrRootOperation = GetStatementOrRootOperation(bindingRoot, cancellationToken);
+            if (statementOrRootOperation == null)
+            {
+                return null;
+            }
 
-            GetBoundNodes(node, out bindableNode, out lowestBoundNode, out highestBoundNode, out boundParent);
-            BoundNode result;
-            switch (options)
+            // we might optimize it later
+            // https://github.com/dotnet/roslyn/issues/22180
+            return statementOrRootOperation.DescendantsAndSelf().FirstOrDefault(o => !o.IsImplicit && o.Syntax == node);
+        }
+
+        private CSharpSyntaxNode GetBindingRootOrInitializer(CSharpSyntaxNode node)
+        {
+            CSharpSyntaxNode bindingRoot = GetBindingRoot(node);
+
+            // if binding root is parameter, make it equal value
+            // we need to do this since node map doesn't contain bound node for parameter
+            var parameter = bindingRoot as ParameterSyntax;
+            if (parameter != null && parameter.Default?.FullSpan.Contains(node.Span) == true)
             {
-                case GetOperationOptions.Parent:
-                    result = boundParent;
-                    break;
-                case GetOperationOptions.Highest:
-                    result = highestBoundNode;
-                    break;
-                case GetOperationOptions.Lowest:
-                default:
-                    result = lowestBoundNode;
-                    break;
+                bindingRoot = parameter.Default;
```

Comment:
Consider `return parameter.Default;` here rather than checking other cases below that appear to be distinct. Same comment for other `bindingRoot = ...;` cases below.